script/main.js

```diff
@@ -190,23 +190,28 @@ class ScoringGuide extends Component {
   }
 }
 
+const debounce = (callback, time = 250, interval) =>
+  ((...args) => {
+    clearTimeout(interval);
+    interval = setTimeout(() => callback(...args), time);
+  })
+
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = getInitialState();
   }
 
   componentDidUpdate() {
-    const url = new URL(location.href);
-
-    const params = new URLSearchParams();
-    for (const [id, value] of Object.entries(this.state)) {
-      const metric = scoring[id];
-      params.set(metric.auditId, value);
-    }
-    url.hash = params.toString();
 
-    history.replaceState(this.state, '', url.toString());
+    // debounce just a tad, as its noisy
+    debounce(_ => {
```

Comment:
seems like this wouldn't work unless you stored `interval` somewhere .... but i assume it must work if you put up a PR.

Reply:
yeah its weird but the interval is kept within the closure. and this debounce just returns another function. so thats why i have a () at the end.